File: authentification/src/main/java/fr/deroffal/bibliotheque/api/authentification/SwaggerConfiguration.java
package fr.deroffal.bibliotheque.api.authentification;

import static springfox.documentation.builders.RequestHandlerSelectors.any;
import static springfox.documentation.builders.RequestHandlerSelectors.withClassAnnotation;
import static springfox.documentation.spi.DocumentationType.SWAGGER_2;

import io.swagger.annotations.Api;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spring.web.plugins.Docket;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

/**
 * Path : /swagger-ui.html
 */
//@formatter:off
@Configuration
@EnableSwagger2
public class SwaggerConfiguration {

    @Bean
    public Docket api() {
        return new Docket(SWAGGER_2)
                .groupName("all")
                .select()
                .apis(withClassAnnotation(Api.class))
                .build();
    }

    @Bean
    public Docket publicApi() {
        return new Docket(SWAGGER_2)
                .groupName("public")
                .select()
                .apis(any())
                .paths(s -> s.contains("/public/"))
                .build();
    }

}

File: livre/src/main/kotlin/fr/deroffal/bibliotheque/livre/adapter/apidocs/SwaggerConfiguration.kt
package fr.deroffal.bibliotheque.livre.adapter.apidocs

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import springfox.documentation.builders.ApiInfoBuilder
import springfox.documentation.builders.PathSelectors
import springfox.documentation.builders.RequestHandlerSelectors
import springfox.documentation.spi.DocumentationType.SWAGGER_2
import springfox.documentation.spring.web.plugins.Docket
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.stream.Collectors.joining

@Configuration
class SwaggerConfiguration {

    @Bean
    fun api(): Docket {
        val description = lireDescription("/apidocs/livre.md")
        val apiInfo = ApiInfoBuilder()
            .title("Bibliothèque - API Livre")
            .description(description)
            .build()
        return Docket(SWAGGER_2)
            .apiInfo(apiInfo)
            .select()
            .apis(RequestHandlerSelectors.any())
            .paths(PathSelectors.any())
            .build()
    }

    private fun lireDescription(emplacement: String) = javaClass.getResourceAsStream(emplacement)
        ?.let {
            BufferedReader(InputStreamReader(it))
                .lines()
                .collect(joining(System.lineSeparator()))
        }

}
